pkg/kafka/policy.go

```diff
@@ -15,13 +15,27 @@
 package kafka
 
 import (
-	"github.com/cilium/cilium/pkg/flowdebug"
 	"github.com/cilium/cilium/pkg/policy/api"
 
 	"github.com/optiopay/kafka/proto"
-	"github.com/sirupsen/logrus"
 )
 
+func matchNonTopicRequests(req *RequestMessage, rule api.PortRuleKafka) bool {
+	// These are no topic requests. If topic is present, we need to return
+	// false since topic can never be associated with this request kind
+	// The topic can be non null if APIRole is specified.
+	// If topic is non null, and so is APIKey, this is due to the policy
+	// correlating a non-topic apikey like apiversions with topic.
+	// We should return false in this case.
+	if rule.Topic != "" && len(rule.APIKey) != 0 {
+		return false
+	}
+	// TODO add functionality for parsing clientID GH-3097
```

Comment:
This will need changes in the optiopay parser, so I have opened a GH issue to do this as a separate PR.